plugins/pluginmanagers/lazy.nix

```
@@ -1,223 +1,187 @@
+      lazyPath =
+        let
+          processPlugin =
+            plugin:
+            [ (if lib.isDerivation plugin then plugin else plugin.pkg) ]
+            ++ lib.optionals ((plugin.dependencies or null) != null) (
+              builtins.concatMap processPlugin plugin.dependencies
+            );
         in
-        mkOption {
-          type = listOfPlugins;
-          default = [ ];
-          description = "List of plugins";
-        };
-    };
-  };
-
-  config = mkIf cfg.enable {
-    extraPlugins = [ pkgs.vimPlugins.lazy-nvim ];
-
-    extraPackages = [ cfg.gitPackage ];
-
-    extraConfigLua =
-      let
-        pluginToLua =
-          plugin:
-          let
-            keyExists = keyToCheck: attrSet: lib.elem keyToCheck (lib.attrNames attrSet);
-          in
-          if isDerivation plugin then
-            { dir = "${lazyPath}/${lib.getName plugin}"; }
-          else
-            {
-              "__unkeyed" = plugin.name;
-
-              inherit (plugin)
-                cmd
-                cond
-                config
-                dev
-                enabled
-                event
-                ft
-                init
-                keys
-                lazy
-                main
-                module
-                name
-                optional
-                opts
-                priority
-                submodules
-                ;
-
-              dependencies = helpers.ifNonNull' plugin.dependencies (
-                if isList plugin.dependencies then (pluginListToLua plugin.dependencies) else plugin.dependencies
-              );
-
-              dir =
-                if plugin ? dir && plugin.dir != null then plugin.dir else "${lazyPath}/${lib.getName plugin.pkg}";
-            };
+        pkgs.linkFarmFromDrvs "lazy-plugins" (builtins.concatMap processPlugin lazyPlugins);
 
-        pluginListToLua = map pluginToLua;
+      setupOptions =
+        let
+          pluginToSpec =
+            plugin:
+            if isDerivation plugin then
+              { dir = "${lazyPath}/${lib.getName plugin}"; }
+            else
+              plugin
+              // {
+                "__unkeyed" = plugin.name;
+                dir = if plugin ? dir && plugin.dir != null then plugin.dir else "${lazyPath}/${plugin.pkg.name}";
+                dependencies = helpers.ifNonNull' plugin.dependencies (
+                  if isList plugin.dependencies then (pluginListToSpecs plugin.dependencies) else plugin.dependencies
+                );
+                # Set pkg property to null as it's been used and no longer needed.
+                pkg = null;
+              };
 
-        plugins = pluginListToLua cfg.plugins;
+          pluginListToSpecs = map pluginToSpec;
 
-        packedPlugins = if length plugins == 1 then head plugins else plugins;
-      in
-      mkIf (cfg.plugins != [ ]) ''
-        require('lazy').setup(
+          pluginSpecsList = pluginListToSpecs lazyPlugins;
```

Sometimes it's better to have the logic together inline :slightly_smiling_face: 

```suggestion
          pluginSpecsList = map pluginToSpec lazyPlugins;
```